test/go/genmock.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/sh
+set -e
+
+export GOPATH=`pwd`
+export GOBIN=`pwd`/bin
+export GO111MODULE=off
+
+mkdir -p src/github.com/golang/mock
+cd src/github.com/golang
+curl -fsSL https://github.com/golang/mock/archive/v1.2.0.tar.gz -o mock.tar.gz
```

Comment:
I'm surprised one cannot pin mock to 1.2.0 using a file similar to node packages.json or pip requirements.txt?

Reply:
One could, but Thrift doesn't use any dependency managers for Go. Prior to `golang/mock` 1.3.0, they also shipped all their dependencies with the repo (the `vendor/` folder) so it wasn't a problem for us.

If we used `dep` (the old Go package manager) or Go modules (the new one) we could pin it easily.
Our current folder structure is making this very difficult, because we are actively working around how Go wants packages to be structured.

I'd like to reorganize it and make it module compatible as soon as I have some free time.